### ISSUE TITLE
Acceptance tests summary comment not added to the PR from forks

### ISSUE THREAD

**Issue by davidgamez** [MobilityData/gtfs-validator#1960]
### Describe the bug

The acceptance tests summary is not added to the PR when the action is triggered in forks.
The following error is visible in the logs:
```
(node:1) UnhandledPromiseRejectionWarning: HttpError: Resource not accessible by integration
    at /node_modules/@octokit/request/dist-node/index.js:66:23
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:1) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
Example: 
- PR -> https://github.com/MobilityData/gtfs-validator/pull/1959
- GitHub action -> https://github.com/MobilityData/gtfs-validator/actions/runs/13128106708/job/36650937663?pr=1959

### Steps/Code to Reproduce

- Create a fork with a non-contributor user
- Create a PR that triggers the acceptance tests

### Expected Results

The acceptance tests summary comment is added to the PR.

### Actual Results

The acceptance tests summary comment is not added and an error is visible in the logs of the `compare-output` action.

### Validator version

N/A

### Operating system

N/A

**Comment by mackenziereading19** [comment]
I traced the current acceptance workflow. It still runs under pull_request, executes the PR code, and then tries to post the generated summary using GITHUB_TOKEN. For fork PRs that token is read-only, which matches the reported Resource not accessible by integration failure.

I don't think simply granting pull-requests: write fixes fork runs, and moving the acceptance workflow itself to pull_request_target would be unsafe because it executes PR-controlled code.

Would you be comfortable with a two-stage approach: keep the acceptance tests on pull_request, upload the generated summary as an artifact, then use a trusted workflow_run workflow with minimal pull-requests: write permission solely to retrieve that artifact and post it to the originating PR? I can implement that with the privileged workflow avoiding checkout/execution of fork code.
